### PR TITLE
refactor cpu pinning

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *Porting [PR 22489](https://github.com/openvinotoolkit/openvino/pull/22489)*
 -- *add ov::affinity deprecation in CPU plugin introduction*
 -- *refactoring get_cpu_pinning()*
 -- *Default CPU pinning on Windows is false on all platform*
 - *Enable CPU pinning in part of CPU functional test cases. Now Intel CPU plugin func tests on Windows finish at 0h:27m:32.21s in this PR and 0h:30m:50.99s in [PR 23129](https://github.com/openvinotoolkit/openvino/pull/23129)*

### Tickets:
 - *CVS-129030*

### CODE REVIEW

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@ void core_configuration(ov::test::SubgraphBaseTest* test) {
         // todo: issue: 123320
         test->convert_precisions.insert({ov::element::bf16, ov::element::f32});
         test->convert_precisions.insert({ov::element::f16, ov::element::f32});
+        test->configuration.insert({ov::hint::enable_cpu_pinning.name(), true});
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:20]: Please add the comment why we need this setting

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
@wangleis Please also update PR description with Win test pass time change (before/after).

**Review by xufang-lisa — APPROVED** [review]

**Review by sunxiaoxia2022 — APPROVED** [review]

**Comment by wangleis** [comment]
> @wangleis Please also update PR description with Win test pass time change (before/after).

Updated

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp (hunk 1 above)
Comment (wangleis) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:20]: updated

Comment (wangleis) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp:23]: updated

**Pull request merged**